app/src/main/java/org/mozilla/fenix/home/HomeFragmentStore.kt

```diff
@@ -144,6 +153,48 @@ private fun homeFragmentStateReducer(
                     }
             )
         }
-        is HomeFragmentAction.PocketArticlesChange -> state.copy(pocketArticles = action.pocketArticles)
+        is HomeFragmentAction.PocketStoriesChange -> state.copy(pocketStories = action.pocketStories)
+        is HomeFragmentAction.PocketStoriesCategoriesChange -> state.copy(
+            pocketStoriesCategories = action.storiesCategories
+        )
+        is HomeFragmentAction.PocketStoriesCategoryClicked -> {
```

Comment:
The event is hoisted from the lowest composable to the ViewHolder where I first had this logic and from where I'd just emit an action with updated data which the Reducer would update in Sate.
But a View Holder shouldn't control the functionality, just bind the Views.
So in the spirit of MVI and everything being a flow of data I went with having the reducer ultimately deciding how the data is updated and updating this in State.
Going further without using Interactors and Controllers maybe this is the architecture to use?

Reply:
Talked with Sebastian about this who said that this is not the responsability of the reducer.
Will move this to the Fragment.

Reply:
Moved.